### PR TITLE
Claim partial investments

### PR DESCRIPTION
**Pull request by nenadV91**
# Summary

If the `Buy vCOW with USDC` option is not set to 100% it throws this, otherwise I think it works. Also not sure what to do if the user selects the investment option but then doesn't change the value from 0. Should we exclude it (at what point in the flow) or not?
Screenshot from 2022-01-20 18-57-39
Screenshot from 2022-01-20 18-57-50

### CODE REVIEW

**Review by alfetopito — APPROVED** [review]
The issue was that there was a bug in the code that built the contract args.

I pushed a fix and managed to claim partial investment of every kind https://rinkeby.etherscan.io/tx/0xe0f162638784590ff0548889f498c0c6592591d3cbab955dccca33d7aaaa30f5

**Comment by alfetopito** [comment]
I'm merging as I'm dealing with code that needs this.

Post merge review etc etc

**Comment by elena-zh** [comment]
Awesome!

File: src/custom/state/claim/hooks/utils.ts [hunk 1]
```diff
@@ -215,3 +217,20 @@ export function calculateInvestmentAmounts(
   const amount = CurrencyAmount.fromRawAmount(currencyAmount.currency, investedAmount)
   return { vCowAmount: price.quote(amount), investmentCost: amount }
 }
+
+/**
+ * Helper function that prepares investFlowData for claiming by calculating vCowAmount from investedAmounts
+ */
+export function prepareInvestClaims(investFlowData: InvestClaim[], userClaimData: EnhancedUserClaimData[]) {
```
Comment (anxolin) [inline src/custom/state/claim/hooks/utils.ts:224]: good call adding a helper function instead of complicating more the tx preparation logic

**Pull request merged**